cudax/include/cuda/experimental/__container/vector.h

```diff
@@ -0,0 +1,426 @@
+  }
+
+  //! @brief Move constructs a \c vector
+  //! @param __other The other vector
+  //! The new vector takes ownership of the allocation of \p __other
+  vector(vector&& __other) noexcept
+      : __buf_(_CUDA_VSTD::move(__buf_))
+      , __size_(__other.__size_)
+  {
+    __other.__size_ = 0;
+  }
+
+  //! @brief Constructs a \c vector with a given size
+  //! @param __mr The memory resource to allocate the vector with.
+  //! @param __size The size of the vector
+  //! @warning This constructor does *NOT* initialize any elements. It is the users responsibility to ensure that the
+  //! elements within `[buf.begin(), buf:begin() + size)` are properly initialized, e.g with `std::uninitialized_copy`
+  vector(_CUDA_VMR::resource_ref<_Properties...> __mr, const size_t __size, uninitialized_construction_t)
+      : __buf_(__mr, __size)
+      , __size_(__size)
+  {
+    _LIBCUDACXX_ASSERT(__size <= __buf.size(),
+                       "vector(buf, size): size is larger than the capacity of the passed buffer");
+  }
+
+  //! @brief Copy assignment
+  //! @param __other The other vector
+  //! @note Even if the old vector would have enough storage available, we have to reallocate if the stored memory
+  //! resource is not equal to the new one.
+  vector& operator=(const vector& __other)
+  {
+    // There is sufficient space in the allocation and the resources are compatible
+    if (resource() == __other.resource() && __buf_.size() == __other.__buf_.size())
+    {
+      if (__size_ >= __other.__size_)
+      {
+        const auto __res = _CUDA_VSTD::copy(__other.begin(), __other.end(), begin());
+        _CUDA_VSTD::__destroy(__res, end());
+      }
+      else
+      {
+        const auto __res = _CUDA_VSTD::copy(__other.begin(), __other.begin() + __size_, begin());
+        _CUDA_VSTD::uninitialized_copy(__other.begin() + __size_, __other.end(), __res);
+      }
+      return *this;
+    }
+
+    // We need to reallocate and copy
+    const uninitialized_buffer<T, _Properties...> __new_buf_{__other.resource(), __other.__size_};
+    _CUDA_VSTD::uninitialized_copy(__other.begin(), __other.end(), __new_buf_.begin());
+
+    // Now that everything is set up bring over the new data
+    _CUDA_VSTD::__destroy(begin(), end());
+    __buf_ = _CUDA_VSTD::move(__new_buf_);
+    return *this;
+  }
+
+  //! @brief Move assignment
+  //! @param __other The other vector
+  vector& operator=(vector&& __other) noexcept
+  {
+    _CUDA_VSTD::__destroy(begin(), end());
+    __buf_  = _CUDA_VSTD::move(__other.__buf_);
+    __size_ = __other.__size_;
+    return *this;
+  }
+
+  //! @brief Destroys the \c vector deallocating the storage after destroying all elements
+  ~vector() noexcept
+  {
+    _CCCL_IF_CONSTEXPR (!_CCCL_TRAIT(is_trivially_destructible, _Tp))
+    {
+      _CUDA_VSTD::__destroy(begin(), end());
+    }
+  }
+  //! @}
+
+  //! @addtogroup iterators
+  //! @{
+  //! @brief Returns pointer to the start of the vector
```

Comment:
Mimic std::vector docs? (cppreference)

```suggestion
  //! @brief Returns an iterator to the first element of the vector. If the vector is empty, the returned iterator will be equal to end().
```

Reply:
Or at least use "iterator", not "pointer".